Dikubot/Webapp/Shared/Facebook/FacebookAddPages.razor

```diff
@@ -0,0 +1,144 @@
+﻿@page "/FacebookAddPages"
+@using System.Net.Http.Json
+@using System.Threading
+@using Dikubot.DataLayer.Database.Global.Facebook
+@inject FacebookPageService _FacebookPageService
+@inject NavigationManager _navigationManager
+
+@attribute [AllowAnonymous]
+<h3>fb</h3>
+
+<MudButton OnClick="LoginFacebookAction" Variant="Variant.Outlined"> <MudIcon Icon="@Icons.Custom.Brands.Facebook"></MudIcon> Facebook Connect</MudButton>
+
+@ResponsFromFacebook
+
+<MudDataGrid Items="@FBresponsList">
+    <Columns>
+        <HierarchyColumn T="FBPages" />
+                <Column T="FBPages">
+                    <CellTemplate>
+                        <MudAvatar Image="@GetFBPicture(context.Item.id)"/>
+                    </CellTemplate>
+                </Column>
+        <Column T="FBPages" Field="name" Title="Name"/>
+    </Columns>
+     <ChildRowContent>
+            <MudCard>
+                <MudDataGrid Items="@FBEvents(context.Item.id)">
+                    <Columns>
+                        <Column T="FBEvent" Field="name"  Title="Name" />
+                        <Column T="FBEvent" Field="start_time"  Title="Start tidspunkt" />
+                        <Column T="FBEvent" Field="end_time"  Title="Slut tidspunkt" />
+                        <Column T="FBEvent" Field="description"  Title="Description" />
+                    </Columns>
+                </MudDataGrid>
+            </MudCard>
+        </ChildRowContent>
+</MudDataGrid>
+
+
+
+@code {
+    
+    [Inject] private HttpClient HttpClient { get; set; }
+    private string? AuthCode { get; set; } = "";
+
+    private string ResponsFromFacebook { get; set; } = "";
+    
+    private List<FBPages> FBresponsList = new List<FBPages>();
+
+    private List<FBEvent> FBEvents(string id)
+    {
+        var link2 = $"https://graph.facebook.com/v16.0/{id}/events?access_token={AuthCode}";
+        return HttpClient.GetFromJsonAsync<FBEventRespons>(link2).Result.data;
+    }
+    
+    private string GetFBPicture(string id)
+    {
+        if (!string.IsNullOrWhiteSpace(AuthCode))
+        {
+            var link = $"https://graph.facebook.com/v16.0/{id}/picture?redirect=0&access_token={AuthCode}";
+            return HttpClient.GetFromJsonAsync<FBPictureRespons>(link).Result.data.url;
+
+    //return ;
+        }
+        return "";
```

Comment:
Most of this should be moved into a service. Consider what I have previously mentioned when it comes to implementing services with dependency injection.

Reply:
Same can be said for a lot of the code under this.